benchmark/misc/fixed-regexp.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const common = require('../common.js');
+const _fastTest = require('_http_outgoing')._fastTest;
+const upgradeExpression = /^Upgrade$/i;
+const upgradeExpressionLength = 'Upgrade'.length;
+const v8 = require('v8');
+
+const bench = common.createBenchmark(main, {
+  type: ['regexp', 'faster'],
+  field: ['Upgrade', 'Upgrade!', 'Connectionbalalalalala'],
+  n: [10e5]
+});
+
+function directTest(input) {
+  return upgradeExpression.test(input);
+}
+
+function fasterTest(input) {
+  return _fastTest(input, upgradeExpression, upgradeExpressionLength);
+}
+
+function main(conf) {
+  let fn;
+  const n = conf.n | 0;
+  const field = conf.field;
+  let v8command;
+
+  if (conf.type === 'regexp') {
+    fn = directTest;
+    v8command = '%OptimizeFunctionOnNextCall(directTest)';
+  } else if (conf.type === 'faster') {
+    fn = fasterTest;
+    v8command = '%OptimizeFunctionOnNextCall(fasterTest)';
+  }
+
+  // Force-optimize the method to test so that the benchmark doesn't
+  // get disrupted by the optimizer kicking in halfway through.
+  for (var i = 0; i < conf.type.length * 10; i += 1)
+    fn(field);
```

Comment:
This loop is unnecessary if you're already calling `%OptimizeFunctionOnNextCall()`.